### PR TITLE
hub/quota: test, maximum, refactoring, ...

### PR DESCRIPTION
**Pull request by haraldschilly**
# Questions
* why is that cpu request divided by 1024? I don't understand that at all. Is somewhere an additional factor at play which I don't see? in any case, I'm confused, and at least initially I'm preserving the previous behavior.
   * hsy: I think this is from code a long time ago. The factor is divided away anyways.

# Description
* this properly caps user contributed upgrades
* this depends on #3359 ... not technically, but semantically
* related to that, settings-quotas for the cpu shares of 256 should be set to zero. then the part in the code removed that subtracts 256 out of nowhere. then it's more consistent.
  * I've not changed that part, but instead let's merge this first and then run the query to fix the `cpu_shares`. Then as a follow up we remove that part of the code.

# Testing Steps
1. this calls for k2

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):
- [ ] No debugging console.log messages.
- [ ] All new code is actually used.
- [ ] Non-obvious code has some sort of comments.

Front end:
- [ ] Restart at least one project and check its `~/.smc/local_hub/local_hub.log`
- [ ] Completely restart Webpack with `./w` in `/src`
- [ ] Screenshots if relevant.

### CODE REVIEW

**Comment by johnjeng** [comment]
What does `PR-pending` mean?

**Comment by haraldschilly** [comment]
they're listed here: https://github.com/sagemathinc/cocalc/labels?utf8=%E2%9C%93&q=pr-

>  "finished, but depends on another branch"

together with work in progress I try to signal that this is not yet ready but close.

**Comment by haraldschilly** [comment]
question for you, maybe, what does "PR-RFC" do? there is one with `[RFC]` in the title, too.  my first guess is "request for comments" ?

**Comment by johnjeng** [comment]
Huh, I don't remember creating the RFC label. But yeah, it's request for comments. Maybe I should just use `needs-help`.

**Comment by johnjeng** [comment]
Oh, wow there are descriptions now. That's great.

File: src/smc-hub/kucalc/quota.coffee [hunk 1]
```diff
@@ -59,63 +88,64 @@ exports.quota = (settings, users) ->
     #            quota.privileged = true
     #            break
 
+    # little helper to caluclate the quotas, contributions, and limits
```
Comment (williamstein) [inline src/smc-hub/kucalc/quota.coffee]: typo: "caluclate"

**Pull request merged**